### PR TITLE
Changes to reduce startup time of Micronaut App

### PR DESCRIPTION
**Pull request by nikhmalh**
We observed that using micronaut-liquibase has blown up the start up time of Micronaut App, even if there are no changes to be made to Database. I have taken up the code from micronaut liquibase library and did some optimizations on it to reduce the startup time of the application, when there are no changes to be made to database. This is because the "performUpdate" method is called irrespective of whether the changes are really required to be made to db. I have created another method "performUpdateIfNeeded()" that checks whether there are unrun changesets or not. If there are no unrun changesets, it won't call "performUpdate" and "liquibase.update" at all.

### CODE REVIEW

File: liquibase/.checkstyle [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (graemerocher) [inline liquibase/.checkstyle:1]: remove this

File: liquibase/src/main/java/io/micronaut/liquibase/LiquibaseMigrationRunner.java [hunk 2]
```diff
@@ -176,6 +179,35 @@ private void migrate(LiquibaseConfigurationProperties config, DataSource dataSou
...
+        .getChangeLogService(liquibase.getDatabase())
+        .reset();
+        performUpdate(liquibase, config);
+      }
+    }
+
+    private static boolean updateNeeded(
+        final Liquibase liquibase, final Contexts contexts, final LabelExpression labelExpression)
+            throws LiquibaseException {
+      LOG.info("Checking if update required...");
```
Comment (graemerocher) [inline liquibase/src/main/java/io/micronaut/liquibase/LiquibaseMigrationRunner.java:205]: change to `debug` level

File: liquibase/src/main/java/io/micronaut/liquibase/LiquibaseMigrationRunner.java [hunk 3]
```diff
@@ -176,6 +179,35 @@ private void migrate(LiquibaseConfigurationProperties config, DataSource dataSou
...
+      final List<ChangeSet> unrunChangeSets =
+          liquibase.listUnrunChangeSets(contexts, labelExpression, false);
+      LOG.info("Size of unrunchangeset : " + unrunChangeSets.size());
```
Comment (graemerocher) [inline liquibase/src/main/java/io/micronaut/liquibase/LiquibaseMigrationRunner.java:208]: Change to

```java
LOG.debug("Size of unrunchangeset : {}", unrunChangeSets.size());
```

**Review by graemerocher — CHANGES_REQUESTED** [review]

File: liquibase/src/main/java/io/micronaut/liquibase/LiquibaseMigrationRunner.java (hunk 3 above)
Comment (nikhmalh) [reply liquibase/src/main/java/io/micronaut/liquibase/LiquibaseMigrationRunner.java]: Log level changed to debug and removed check style file.

**Comment by alvarosanchez** [comment]
Superseded by #279

**Pull request closed without merging**